nativelib/src/main/scala/scala/scalanative/runtime/NativeExecutionContext.scala

```diff
@@ -27,21 +58,34 @@ object NativeExecutionContext {
 
     override def reportFailure(t: Throwable): Unit = t.printStackTrace()
 
-    def hasNextTask: Boolean = !queue.isEmpty
-    def availableTasks: Int = queue.size
+    override def stealWork(maxSteals: Int): Unit = {
+      if (maxSteals <= 0) ()
+      else {
+        var steals = 0
+        while (steals < maxSteals && hasAvailableTasks) {
+          doStealWork()
+          steals += 1
+        }
+      }
+    }
```

Comment:
```suggestion
    override def stealWork(maxSteals: Int): Boolean = {
      if (maxSteals > 0) {
        var steals = 0
        while (steals < maxSteals && hasAvailableTasks) {
          doStealWork()
          steals += 1
        }
      }
      !queue.isEmpty
    }
```
like this allows to implement libuv integration without any issue

Reply:
Maybe it's the issue with taking look at it at 2 AM, but I don't really get how returning boolean helps here. Wouldn't having an additional call to public method returning the same value would be feasible? Especially in context of previous snippet with libuv usage seems to confirm that, becouse I assume it could be rewritten as following I suppose: 
```scala
  val queue = QueueExecutionContext
  private def loopUV(): Boolean = {
    queue.stealWork(1) // execute first or noop
    if (queue.hasAvailableTasks) {
      if (uv_loop_alive(uvLoop) != 0) {
        uv_update_time(uvLoop)
        uv_run(uvLoop, uv_run_mode.NOWAIT)
      }
    } else if (uv_loop_alive(uvLoop) != 0) uv_run(uvLoop, uv_run_mode.ONCE)
    (uv_loop_alive(uvLoop) != 0) || queue.hasAvailableTasks
  }
```
Disclaimer: I do miss context about usage of libuv, I have no experience in it and don't really know how you're integrating it in rest of the code

Reply:
Yes, making `hasAvailableTasks` public helps here.

Regarding original issue or some context: `libuv` is just an example event-loop. You may think about it as a `pool` or `select`. The goal is mix `poll`/`select` with scala-native event loop with trivial logic:
 - if SN has something else, run multiplexor without blocking to fetch that it has already;
 - if SN hasn't got anything else, block on it until something happened.